src/Commands/MakeFactoryCommand.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace KoalaFacade\DiamondConsole\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Filesystem\FileNotFoundException;
+use Illuminate\Support\Str;
+use KoalaFacade\DiamondConsole\Actions\Filesystem\FilePresentAction;
+use KoalaFacade\DiamondConsole\Actions\Stub\CopyStubAction;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasArguments;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasOptions;
+use KoalaFacade\DiamondConsole\Commands\Concerns\InteractsWithPath;
+use KoalaFacade\DiamondConsole\DataTransferObjects\CopyStubData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\Filesystem\FilePresentData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\PlaceholderData;
+
+class MakeFactoryCommand extends Command
+{
+    use InteractsWithPath, HasArguments, HasOptions;
+
+    protected $signature = 'diamond:factory {name} {domain} {--force}';
+
+    protected $description = 'create new enum';
+
+    /**
+     * @throws FileNotFoundException
+     */
+    public function handle(): void
+    {
+        $this->info(string: 'Generating factory & interface file to your project');
+
+        $this->resolveGenerateForFactoryInterface();
+
+        $this->resolveGenerateForFactoryConcrete();
+    }
+
+    /**
+     * @throws FileNotFoundException
+     */
+    protected function resolveGenerateForFactoryInterface(): void
+    {
+        $fileName = 'Abstract' . $this->resolveNameArgument() . '.php';
+
+        $namespace = $this->resolveNamespace(
```

Comment:
should be into Domain/Shared/Contracts/Database/Factories/{Model.php} this one will `extends` Domain/Shared/Contracts/Database/Factory.php

// Factory.php
```
interface Factory
{
    /**
     * Create a new factory instance for the model.
     *
     * @return EloquentFactory
     */
    public function resolveFactory(): EloquentFactory;
}
```

// Model.php
interface User extends Factory
{
}